tests/test_grains.py

```diff
@@ -914,3 +989,2290 @@ def test_keep_largest_labelled_region(
+                            [0, 1, 0, 0, 1, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 0, 0, 0, 0, 0],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 1, 1, 0, 0],
+                            [0, 0, 1, 1, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                        ]
+                    ),
+                ],
+                axis=-1,
+            ),
+            [(1, 2)],
+            np.stack(
+                [
+                    np.array(
+                        [
+                            [1, 1, 1, 1, 1, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 1, 1, 1, 1, 1],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 1, 1, 0, 1, 0],
+                            [0, 1, 0, 0, 1, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 0, 0, 0, 0, 0],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 1, 1, 0, 0],
+                            [0, 0, 1, 1, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 0, 0, 0, 0, 0],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [0, 0, 0, 0, 0, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 1, 1, 1, 1, 0],
+                            [0, 0, 0, 0, 0, 0],
+                        ]
+                    ),
+                ],
+                axis=-1,
+            ),
+        )
+    ],
+)
+def test_merge_classes(
+    grain_mask_tensor: npt.NDArray[np.int32],
+    classes_to_merge: list[tuple[int, int]],
+    expected_result_grain_mask_tensor: npt.NDArray[np.int32],
+) -> None:
+    """Test the merge_classes method of the Grains class."""
+    result_grain_mask_tensor = Grains.merge_classes(grain_mask_tensor, classes_to_merge)
+
+    np.testing.assert_array_equal(result_grain_mask_tensor, expected_result_grain_mask_tensor)
+
+
+@pytest.mark.parametrize(
+    (
+        "grain_mask_tensor",
+        "pixel_to_nm_scaling",
+        "class_size_thresholds",
```

Comment:
Tests fail as there are only nine parameter names here but there are ten parameters in each `pytest.param()`.

L:eyes:king at the arguments for the function and the comments within the parameters themselves I think its this...

```suggestion
        "class_conversion_size_thresholds",
        "class_size_thresholds",
```